### DISCUSSION TITLE
Welcome to shopping-time Discussions!

### DISCUSSION THREAD

**Discussion by catherineisonline** [catherineisonline/shopping-time#6] [Announcements]
Welcome to shopping-time Discussions! This is a place where you can share your ideas, suggestions, questions, or anything interesting!

# Rules to follow

## Respect and Professionalism:
Be respectful and professional in all interactions.
Avoid offensive, discriminatory, or harmful language and behavior.

## Stay On Topic:
Keep discussions relevant to the project's goals and objectives.
If you have unrelated issues or questions, create a separate issue.

## Search Before Posting:
Search existing issues and discussions before creating a new one to avoid duplication.
Contribute to existing discussions when appropriate.

## Provide Context:
Clearly state your problem or question, providing sufficient context and details.
Include error messages, screenshots, or relevant code snippets as needed.

## One Issue Per Thread:
Keep each discussion or issue focused on a single topic or problem.
This helps to maintain clarity and organization.

## Be Constructive:
Offer constructive feedback and suggestions.
Avoid simply criticizing without providing solutions or alternatives.

## Use Markdown Effectively:
Format your comments and discussions using Markdown for readability.
Use code blocks, bullet points, and headers as needed.

## No Spam or Self-Promotion:
Do not use the project's discussions for spamming or self-promotion.
Avoid sharing links to unrelated content.

## Be Patient:
Understand that maintainers and contributors have their own schedules.
Be patient while waiting for responses or issue resolutions.

## Respect Maintainer Decisions:
Accept the final decisions made by project maintainers regarding issue resolution and project direction.
If you disagree, express your concerns respectfully.

## No Personal Attacks:
Do not engage in personal attacks or harassment.
Focus on the technical aspects of the discussion.

## Contributor Guidelines:
Follow any specific contributor guidelines or code of conduct provided by the project.
Respect the project's licensing and contribution policies.

## Language and Tone:
Communicate in a language that is widely understood within the project community.
Maintain a friendly and welcoming tone.

## Community Moderation:
Encourage community members to help support discussions and report any violations of these rules.

**_Happy coding!_**

**Comment by dev-hariom** [comment]
how we can access 'products' and 'currencies' database. or how i can create. @catherineisonline please help

**Reply by catherineisonline** [reply]
Hello. I store everything in the Firestore, which is my private account, so due to free tiers, I can't make it public. 

However, you can create your own Firebase account, create the database for the products, and then connect to the app with your own API keys. I don't have the schema unfortunatly but I will attach the screenshots for you to create the structure that matches it.  I will try to add the schema later so it's easier. 

You can also try to paste the source code to AI and ask it to create the schema based on the usage, though it might need some manual validation.

<img width="1071" height="578" alt="5" src="https://github.com/user-attachments/assets/7b4e23dc-3480-4c19-8ea6-7f0b08d82f25" />
<img width="1079" height="584" alt="6" src="https://github.com/user-attachments/assets/527e0289-c885-402b-8864-31b6089294cf" />
<img width="1072" height="560" alt="1" src="https://github.com/user-attachments/assets/e267747c-c9e6-4456-a42b-600abf5befe7" />
<img width="1065" height="558" alt="2" src="https://github.com/user-attachments/assets/2908b3fb-c0b7-452c-8db1-624334d5a2c6" />
<img width="1065" height="561" alt="3" src="https://github.com/user-attachments/assets/0b5479cc-74e0-4bbb-812b-7f69fdedc71c" />
<img width="1073" height="567" alt="4" src="https://github.com/user-attachments/assets/d80dbc25-0495-44e6-b510-81b4b951ad43" />

**Reply by catherineisonline** [reply]
Here is a quick schema but I haven't verified whether it's 100% correct:

```
// CATEGORY
Category {
  id: string
  name: string
}

// CURRENCY
Currency {
  label: string
  symbol: string
}

// PRODUCT
Product {
  id: string
  name: string
  brand: string
  category: string
  inStock: boolean
  description: string
  gallery: string[]

  prices: {
    amount: number
    currency: Currency
  }[]

  attributes: {
    id: string
    items: {
      id: string
      displayValue: string
      value: string
    }[]
  }[]
}
```

**Reply by dev-hariom** [reply]
thank you so much @catherineisonline . it helped a lot, very much appreciated.

**Reply by catherineisonline** [reply]
> thank you so much @catherineisonline . it helped a lot, very much appreciated.

Glad to hear that
